src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java

```diff
@@ -267,18 +269,28 @@ public Response remove(String id, String rev) {
      */
     public List<Response> bulk(List<?> objects, boolean allOrNothing) {
         assertNotEmpty(objects, "objects");
-        InputStream response = null;
+        InputStream responseStream = null;
+        HttpConnection connection;
         try {
             final String allOrNothingVal = allOrNothing ? "\"all_or_nothing\": true, " : "";
             final URI uri = buildUri(getDBUri()).path("_bulk_docs").build();
             final String json = String.format("{%s%s%s}", allOrNothingVal, "\"docs\": ", getGson
                     ().toJson(objects));
-            response = couchDbClient.post(uri, json);
-            return getResponseList(response, getGson(), Response.class,
+            connection = couchDbClient.post(uri, json, HttpConnection.class);
```

Comment:
Whilst I understand the sentiment of trying to have common code for this given that we only have one use of the new `CouchDbClient.post(...)` method you added I think it might be better if that didn't take a type and just returned the connection. Alternatively it might be easier just to make the `HttpConnection` and call execute:
```java
connection = Http.POST(uri, "application/json");
//set content
couchDbClient.execute(connection);
```